### DISCUSSION TITLE
Access .env or config from pinia store

### DISCUSSION THREAD

**Discussion by melokki** [nuxt/nuxt#16519] [Questions] [answered]
Hello
I am trying to access the `publicRuntimeConfig` key from `nuxt.config.ts` file inside a pinia store and nothing works so far.
There is a way to access the config or the `.env` variables from a pinia store?

Thank you

**Comment by yinfoyuan** [comment]
https://v3.nuxtjs.org/docs/usage/runtime-config#runtime-config

**Reply by melokki** [reply]
I already tried this and it's not working. I am getting `nuxt instance is null`

**Comment by melokki** [comment, marked as answer]
I managed to solve this by using the `useRuntimeConfig` inside state like this

```ts
state: () => {
    const { apiBaseUrl } = useRuntimeConfig();

    return {
      data: [],
    };
  },
```

**Reply by PsclDev** [reply]
this seems no longer to work? or at least it is shown undefined for me

**Reply by melokki** [reply]
there were some changes in the `nuxt.config.ts` file

this is how my `nuxt.config.ts` file looks like

```
runtimeConfig: {
    public: {
      apiBaseUrl: process.env.API_BASEURL,
    },
  },
```

and in my state I am using it like this
```
 state: () => {
    baseURL = useRuntimeConfig().public.apiBaseUrl;
```

**Reply by sefacndmr01** [reply]
Error: nuxt instance unavailable :(

**Reply by Kern046** [reply]
It works but you mustn't import useRuntimeConfig in the file manually

**Comment by masud-jatri** [comment]
you can also use this.$nuxt.$config.public.<ENV_VARIABLE>
